### PR TITLE
add a bridge-router-bridge test with router implemented as a namespace

### PR DESCRIPTION
**Pull request by yonghong-song**
This is to implement brb with bcc.

Brenden, ethernet->dst (probably other mac address extraction as well) does not work.
Please investigate. At this point, I use load_half/load_word as the workaround.

Signed-off-by: Yonghong Song <yhs@plumgrid.com>

### CODE REVIEW

File: tests/cc/test_brb.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+    def setup_router_ns(self, ns, veth1_in, veth1_out, veth2_in, veth2_out):
+        subprocess.call(["ip", "netns", "add", ns])
+        subprocess.call(["ip", "link", "add", veth1_in, "type", "veth", "peer", "name", veth1_out])
+        subprocess.call(["ip", "link", "set", veth1_in, "netns", ns])
+        subprocess.call(["ip", "netns", "exec", ns, "ip", "link", "set", veth1_in, "name", "eth0"])
+        subprocess.call(["ip", "link", "add", veth2_in, "type", "veth", "peer", "name", veth2_out])
+        subprocess.call(["ip", "link", "set", veth2_in, "netns", ns])
+        subprocess.call(["ip", "netns", "exec", ns, "ip", "link", "set", veth2_in, "name", "eth1"])
+        subprocess.call(["ifconfig", veth1_out, "up"])
```
Comment (4ast) [inline tests/cc/test_brb.py]: pls use 'ip link set dev up' instead of ifconfig

File: tests/cc/test_brb.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+        # get vm mac address
+        self.vm1_mac = commands.getoutput('ip netns exec ' + self.ns1 + ' cat /sys/class/net/eth0/address')
+        self.vm2_mac = commands.getoutput('ip netns exec ' + self.ns2 + ' cat /sys/class/net/eth0/address')
+
+        # load the program and configure maps
+        self.config_maps()
+
+    def test_brb(self):
+        # our bridge is not smart enough, so send arping for router learning to prevent router
```
Comment (4ast) [inline tests/cc/test_brb.py]: would be good to do an example with real linux bridge

File: tests/cc/test_brb.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+        self.config_router_ns(self.ns_router, self.vm1_rtr_ip, self.vm2_rtr_ip)
+
+        # get vm mac address
+        self.vm1_mac = commands.getoutput('ip netns exec ' + self.ns1 + ' cat /sys/class/net/eth0/address')
+        self.vm2_mac = commands.getoutput('ip netns exec ' + self.ns2 + ' cat /sys/class/net/eth0/address')
+
+        # load the program and configure maps
+        self.config_maps()
+
+    def test_brb(self):
```
Comment (4ast) [inline tests/cc/test_brb.py]: may be include example 'make test' output in commit log and/or this .py file, so users know what to expect?

File: tests/cc/test_brb.py (hunk 2 above)
Comment (yonghong-song) [inline tests/cc/test_brb.py]: Yes, this will be my next step.

**Pull request closed without merging**